### DISCUSSION TITLE
Integration issue and max trunc error

### DISCUSSION THREAD

**Discussion by adamhb** [NGEET/fates#1097] [Q&A]
I recently got an integration error "Plant Growth Integrator could not find a solution in less than 300 tries." Prior [issues](https://github.com/NGEET/fates/issues/352) indicate that you can solve this by increasing the `max_trunc_error` param. In 2018, @rgknox [raised](https://github.com/NGEET/fates/pull/391/commits/0b6261d067efd526ef9da12e120c5f68d2ee8157) the `max_trunc_error` parameter from 1 to 1000, but the [latest version of the code has it back to 1](https://github.com/NGEET/fates/blob/main/parteh/PRTAllometricCarbonMod.F90#L403). Should I try raising this again to 1000 in my version of the code to fix the issue or does doing this cause other issues?

**Comment by rgknox** [comment]
I took a look at the code and refreshed my memory. So this error, for the euler integrator, has units of kgC/plant. And its testing to see if each organ during the growth flux is on allometry, by evaluating the difference between the actual mass of the organ and the allometry value. 
One thing to keep in mind, is that the algorithm is self correcting. SO if it does a poor job at keeping the plant on allometry during the current attempt, it will get back on track in the next step. For instance, if the actual mass is above the allometric target, it will ignore that organ on the next growth step until the other organs are caught up. If the actual mass is below the target, it will have a chance to gain carbon during the phase 1 allocation (getting to targets) before it tries to grow out stature.
Ideally, we want the integration to be right on track if possible, but my sense is that its ok if you increase that max_trunc_error, maybe try 10 first?

**Comment by adamhb** [comment]
OK, thanks @rgknox. I'll give that a shot!

**Comment by adamhb** [comment]
Hi @rgknox 
I had already started some simulations with `max_trunc_error` set to 1,000 before you suggested starting with 10. The error went away, but in some ensemble members I'm getting step changes in basal area for pine and fir (see attached). Do you think this could be related to having a `max_trunc_error` set too high?
Adam
![fir_pine_step_change](https://github.com/NGEET/fates/assets/25392545/aa7d9c06-fe3a-4242-9cd4-cac2b0dc561a)

**Reply by adamhb** [reply]
Oh, I'm seeing that these step changes occur even for runs where the max_trunk_error param is set to 1, so I assume this is not related.

**Reply by adamhb** [reply]
I assume this is happening at the points where BA is jumping up... does that sound right?

![image](https://github.com/NGEET/fates/assets/25392545/8ebdb1fd-7231-4c72-8ec7-1a14bef3de74)

**Reply by rgknox** [reply]
I suppose its possible. Could you try a similar run with higher max cohort counts?  I wonder if this could also be a correction that happens from fusion.  

Or could this have something to do with degraded branch biomass or a change in damage state? For instance, if the plant suddenly changes to a higher damage class, but has not lost any biomass, it would theoretically have to increase its diameter to accommodate a larger bole. 

Can we determine if the steps are happening over the course of one day, or the course of the year?  Or can we rule out some competition dynamic is involved? I noticed the firs had a great year (a step) when the oaks stagnated.

**Reply by adamhb** [reply]
Hi Ryan,
Thanks so much for these thoughts and ideas on what to try. The steps are happening on a shorter timescale than a year, but I'm not sure if they are happening from one day to the next. I am outputting monthly history files and I see the step occur from one month to the next. I didn't turn damage on, but I suppose if its on by default then it could have something to do with damage. I'll try running with higher max cohorts and see if it changes anything. I'll do some tests and let you know if I can get any more insights.
Adam

**Reply by rgknox** [reply]
Are these re-sprouters?

**Reply by adamhb** [reply]
That was my first thought too, but no, only oak and shrub can resprout, but pine, cedar, and fir can't and its happening in the non-resprouting pfts too.
